### PR TITLE
Added Java file dz-12'

### PR DESCRIPTION
**Pull request by annies23**

### CODE REVIEW

File: dz-12/src/test/java/cc/robotdreams/test/ManTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        public void testGetGender(Man man, Person.Gender expectedGender) {
+        Assert.assertEquals(man.getGender(), expectedGender);
+    }
+
+        @DataProvider(name = "retiredManProvider")
+        public static Object[][] provideRetiredManData() {
+        return new Object[][]{
+                {new Man("Сем", "Сміт", 63, null), false},
+                {new Man("Джон", "МакКейн", 66, null), true},
+        };
```
Comment (yurii-voronenko) [inline dz-12/src/test/java/cc/robotdreams/test/ManTest.java:25]: AS I wrote for previous home work, use boundary values: 64, 65, 66

File: dz-12/src/test/java/cc/robotdreams/test/WomanTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        Assert.assertEquals(woman.getGender(), expectedGender);
+    }
+
+    @DataProvider(name = "retiredWomanProvider")
+    public static Object[][] provideRetiredWomanData() {
+        return new Object[][]{
+                {new Woman("Крістіна", "Річі", 58, null), false},
+                {new Woman("Різ", "Візерспун", 61, null), true},
+                // Додайте інші дані, якщо потрібно
+        };
```
Comment (yurii-voronenko) [inline dz-12/src/test/java/cc/robotdreams/test/WomanTest.java:28]: Boundary values for woman are: 59, 60, 61

**Review by yurii-voronenko — APPROVED** [review]
LGTM
